_posts/2019-03-18-rules-jvm-external-maven.md

```diff
@@ -0,0 +1,142 @@
+projects that depend on Maven repositories. The specification of artifacts in
+the repository rule resembles the specification in the `<dependencies>` tag in a
+Maven POM file or the `dependencies` closure in a Gradle build file. In
+particular, this means that users only need to specify their direct dependencies
+and the rule will handle the rest for them.
+
+## Features and use cases
+
+For the initial release, we are targeting support for Java and Android Java
+projects. The following features are supported:
+
+* Supported on Windows, macOS and Linux
+* JAR and AAR packaging types
+* In-band transitive closure resolution
+* Dependency specification directly in the WORKSPACE file
+* Custom Maven repositories
+* Maven repository authentication with HTTP Basic Authentication
+* Artifact version pinning and resolution
+* Multiple independent closures of versioned artifacts
+    * Not forced to only have one version of an artifact per workspace, although
+      different versions are namespaced by the repository name
+* Source JAR support
+* Dependency exclusion
+
+As part of this, we will no longer be maintaining generate_workspace.
+gmaven_rules, which was designed to be an interim solution, will also be
+replaced by rules_jvm_external. We will be updating existing documentation to
+reflect this over the coming weeks.
+
+## How to use
+
+To start using this feature in your Java or Android project, point to the
+repository rule in your WORKSPACE file:
+
+```python
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+http_archive(
+    name = "rules_jvm_external",
+    strip_prefix = "rules_jvm_external-1.1",
+    sha256 = "ade316ec98ba0769bb1189b345d9877de99dd1b1e82f5a649d6ccbcb8da51c1f",
+    url = "https://github.com/bazelbuild/rules_jvm_external/archive/1.1.zip"
+)
+```
+
+Then, specify your direct dependencies using the `artifacts` attribute and
+indicate where those artifacts should come from using the `repositories`
+attribute:
+
+```python
+load("@rules_jvm_external//:defs.bzl", "maven_install")
+
+maven_install(
+    name = "maven",
+    artifacts = [
+        "androidx.test.espresso:espresso-core:3.1.1",
+        "com.google.guava:guava:27.0-android",
+    ],
+    repositories = [
+        "https://maven.google.com",
+        "https://repo1.maven.org/maven2",
+    ],
+    fetch_sources = True,   # Fetch source jars. Defaults to False.
+)
+```
+
+The `maven_install` rule creates targets you can then depend on in your `BUILD`
+files. For example:
+
+```python
+android_library(
+    name = "foo",
+    srcs = ["MyAndroidLib.java"],
+    deps = [
+        "@maven//:androidx_test_espresso_espresso_core"
+        "@maven//:com_google_guava_guava",
+    ],
+)
+```
+
+## How it works
+
+This rule relies on existing tooling to calculate the transitive closure of
+dependencies. The magic underneath is the [Coursier](https://get-coursier.io/)
+CLI. There is a thin Starlark wrapper that translates the input from the rule
```

Comment:
This adds a lot of complexity to the post. What about replacing it with a user-focused section about what they can do with this rule set? Rather than explaining implementation, you could give examples of the scenarios it could handle, explaining that this each scenario is powered by Coursier. What do you think?

It feels like the post is missing this area anyway, so this could kill two birds with one stone. (Filling out the post with more examples and also giving credit to Coursier.)

Reply:
You don't think there might be interest in the implementation? 

As for what you think is missing: could you elaborate? Which scenarios would be helpful to cover that weren't covered by the example above?